#164047064 Build out REST Endpoints for Articles (CRUD)

winniekariuki opened the pull request:
#### What does this PR do?
This pr creates,retrieves,updates and delete articles.

#### Available Endpoints
| Method |          Endpoint                              | Description                                   | Permisssions          |
| ------    | ------------------------------- | --------------------------------| ------------------  |
| POST    | /api/articles                                  | post an article                               | authenticated user|
| GET       | /api/articles/                                |get all articles                                 |any user                  |
| GET       |/api/articles/<char:slug>/            | Get a specific article                     | any user                 |
| PUT       | /api/articles/<char:slug>/           | update an article                            |authenticated user|
| DELETE | /api/articles/<char:slug>/           | delete a specific article                 |authenticated user|
           
#### How should this be manually tested?
After cloning this repo,cd into it and run ```python manage.py runserver``` and test all the above endpoints using postman.
#### What are the relevant pivotal tracker stories?
#164047064

Commit:
164047064-feature(article): Implements CRUD for article

- adds CRUD on article method
- add tests to test the article method
- adds cloudinary

[Finishes #164047064]

JumaKahiga commented:
Kindly adjust views to APIView or Generics for consistency with the existing codebase.

JoanNgatia commented on .travis.yml:
```diff
@@ -8,6 +8,8 @@ install:
   - pip install -r requirements.txt
   - pip install coveralls
 
+addons:
```
We should only need the one `addons` value here

kelvinndmo replied:
@JoanNgatia sometimes we got an error on Travis which was as a result of Travis building with postgres 9.2 which is no longer supported.To solve this you have to specify the postgres version and that is what addons does.

JoanNgatia commented on .env.example:
```diff
@@ -28,3 +28,11 @@ export CLOUDINARY_URL='your cloudinary url goes here'
 export SENDGRID_API_KEY='API_KEY here' #This is the sendgrid api key
 export EMAIL_FROM="test@test.com" # This is the email that is sending the email
 export URL="http://127.0.0.1:8000/" #The base url where the frontend app is hosted
+
```
You'll need to remove the duplicates from this file

Commit:
remove duplicated code

Kibetchirchir committed:
164047064-feature(article): Implements CRUD for article

- adds CRUD on article method
- add tests to test the article method
- adds cloudinary

[Finishes #164047064]
